File: JS/완성/기능개발/기능개발.js
function solution(progresses, speeds) {
  const K = [];
  for (let i = 0; i < progresses.length; i++) {
    let t = Math.ceil((100 - progresses[i]) / speeds[i]);
    K.push(t);
  }

  const P = [K[0]];
  for (let i = 1; i < progresses.length; i++) {
    if (P[P.length - 1] < K[i]) {
      P.push(K[i]);
    } else "";
  }
  console.log(K);
  console.log(P);

  const AN = [];
  if (P.length == 1) {
    AN.push(K.length);
  }
  for (let i = 1; i < P.length; i++) {
    if (P.length == 2) {
      AN.push(K.indexOf(P[1]));
      AN.push(K.length - K.indexOf(P[1]));
    } else if (i == 1) {
      AN.push(K.indexOf(P[1]));
    } else if (i == P.length - 1) {
      AN.push(K.indexOf(P[i]) - K.indexOf(P[i - 1]));
      AN.push(K.length - K.indexOf(P[P.length - 1]));
    } else {
      AN.push(K.indexOf(P[i]) - K.indexOf(P[i - 1]));
    }
  }
  console.log(AN);

  var answer = AN;
  return answer;
}

File: Py/기능개발.py
# https://programmers.co.kr/learn/courses/30/lessons/42586?language=python3

def solution(progresses, speeds):
    day = 1
    answer = []
    count = 0
    for i in range (0,len(progresses)):
        if progresses[i]+speeds[i]*day>=100:
            count = count + 1
        else:
            if i !=0:
                answer.append(count)
            count = 1
            while progresses[i]+speeds[i]*day<100:
                day = day + 1
    answer.append(count)
    return answer
